app/controllers/users/sessions_controller.rb

```diff
@@ -0,0 +1,33 @@
+class Users::SessionsController < Devise::SessionsController
+  respond_to :json
+
+  private
+
+  def respond_with(_resource, _options = {})
+    render json: {
+      status: {
+        code: 200,
+        message: 'User logged in successfully.',
+        data: current_user
+      }
+    }, status: :ok
+  end
+
+  def respond_to_on_destroy
+    jwt_payload = JWT.decode(request.headers['Authorization'].split.last,
+                             Rails.application.credentials.fetch(:secret_key_base)).first
+    current_user = User.find(jwt_payload['sub'])
```

Comment:
Kindly consider removing these lines and replacing `current_user` with something like `@logged_out_user`, and adding the below in your controller.
```ruby
def destroy
  @logged_out_user = current_user
  super
end
```
This approach provides clean and clear code as the `authentication token` may change its definition, from being an `id` to being an `object`, and if that happens you would have to adjust your current logic.